src/checkstyle/Checker.hx

```diff
@@ -246,7 +250,23 @@ class Checker {
 		}
 	}
 
-	@SuppressWarnings("checkstyle:Dynamic")
+	function checkForExclude(moduleName:String):Bool {
+		if (excludes == null) return false;
+		var excludesForCheck:Array<String> = excludes.get(moduleName);
+		if (excludesForCheck == null || excludesForCheck.length == 0) return false;
+
+		var cls = file.name.substring(0, file.name.indexOf(".hx"));
```

Comment:
I'm not sure it should be specific to extensions, it might be more flexible to apply a regex to the entire path. OpenFL generates .hx file in the export directory, which makes it difficult for me to use checkstyle in say, flixel-demos, which has 75 projects.

For generating API docs, I have excluded these specific file names / packages for dox with a regex: https://github.com/HaxeFlixel/flixel-docs/blob/master/api/dox-gen/gendocs.bat

```haxe
--exclude "(__ASSET__|ApplicationMain|DocumentClass|DefaultAssetLibrary|Main|NMEPreloader|zpp_nape)"
```